lib/http_aws_sigv4.c

```diff
@@ -105,6 +188,9 @@ CURLcode Curl_output_aws_sigv4(struct Curl_easy *data, bool proxy)
   unsigned char tmp_sign0[32] = {0};
   unsigned char tmp_sign1[32] = {0};
   char *auth_headers = NULL;
+  char *encoded_path = NULL;
+  size_t encoded_len;
+  char *encoded_query_str = NULL;
```

Comment:
What do you think about renaming vars to: _canonical_uri_ and _canonical_query_str_ ?